python/build_ceed_cffi.py

```diff
@@ -13,6 +13,31 @@
 ceed_version_ge = re.compile(r'\s+\(!?CEED_VERSION.*')
 
 
+# Checks to see if a c line is part of the lines we have to exclude (macros)
+def is_valid_line(line):
+    if (line.startswith("#") and not line.startswith("#include")):
+        return False
+    if (line.startswith("  static")):
+        return False
+    if (line.startswith("  CEED_QFUNCTION_ATTR")):
+        return False
+    if (line.startswith("    return name##_rs")):
+        return False
+    if (line.endswith('\\\n')):
+        return False
+    if (line.startswith('  __NOP')):
+        return False
```

Comment:
```suggestion
    if (line.startswith("  }")):
        return False
    if (line.endswith('\\\n')):
        return False
```
with these, I think you can change the macro def to
```
#define CEED_QFUNCTION_RUST(name)                                                                                            \
  static const char name##_loc[] = __FILE__ ":" #name;                                                                       \
  CEED_QFUNCTION_ATTR int        name##_rs(void *ctx, const CeedInt Q, const CeedScalar *const *in, CeedScalar *const *out); \
  CEED_QFUNCTION_ATTR static int name(void *ctx, const CeedInt Q, const CeedScalar *const *in, CeedScalar *const *out) {     \
    return name##_rs(ctx, Q, in, out);                                                                                       \
  }
#endif
```

Reply:
```suggestion
    if (line.endswith('\\\n')):
        return False
```
I think these two cases aren't needed now though